src/Models/_OfflineModelBase.js

```diff
@@ -0,0 +1,215 @@
+    });
+    return def.promise;
+  },
+  updateEntry: function updateEntity(entry, options) {
+    const store = this.getStore();
+    const def = new Deferred();
+    const entityId = this.getEntityId(entry, options);
+    this.getEntryDoc(entityId).then(function querySuccess(doc) {
+      const odef = def;
+      doc.entity = entry;
+      doc.modifyDate = moment().toDate();
+      doc.description = this.getEntityDescription(entry);
+      store.put(doc).then(function updateSuccess(result) {
+        odef.resolve(result);
+      }, function updateFailed(err) {
+        odef.reject('error updating entity: ' + err);
+      });
+    }.bind(this), function queryError(err) {
+      def.reject('entity not found to update:' + err);
+    });
+    return def.promise;
+  },
+  createEntry: function createEntry() {
+    const entry = {}; // need to dynamicly create Properties;
+    entry.Id = null;
+    entry.CreateDate = moment().toDate();
+    entry.ModifyDate = moment().toDate();
+    return entry;
+  },
+  deleteEntry: function deleteEntry(entityId) {
+    const def = new Deferred();
+    const store = this.getStore();
+    store.get(entityId).then((doc) => {
+      const odef = def;
+      store.remove(doc._id, doc._rev).then((result) => {
+        odef.resolve(result);
+      }, (err) => {
+        odef.reject(err);
+      });
+    }, (err) => {
+      def.reject(err);
+    });
+    return def.promise;
+  },
+  saveRelatedEntries: function saveRelatedEntries(parentEntry, options) {
+    const entries = (parentEntry && parentEntry.$relatedEntities) ? parentEntry.$relatedEntities : [];
+    const relatedPromises = [];
+    const def = new Deferred();
+    entries.forEach((related) => {
+      const model = App.ModelManager.getModel(related.entityName, MODEL_TYPES.OFFLINE);
+      if (model && related.entities) {
+        related.entities.forEach((relatedEntry) => {
+          const promise = model.saveEntry(relatedEntry, options);
+          relatedPromises.push(promise);
+        });
+      }
+    });
+    if (relatedPromises.length > 0) {
+      all(relatedPromises).then(
+          (relatedResults) => {
+            def.resolve(relatedResults);
+          },
+          (err) => {
+            def.reject(err);
+          });
+    } else {
+      def.resolve(parentEntry);
+    }
+    return def.promise;
+  },
+  wrap: function wrap(entry) {
+    let doc;
+    doc = {
+      _id: this.getDocId(entry),
+      entity: entry,
+      entityId: this.getEntityId(entry),
+      createDate: moment().toDate(),
+      modifyDate: moment().toDate(),
+      resourceKind: this.resourceKind,
+      description: this.getEntityDescription(entry),
+      entityName: this.entityName,
+      entityDisplayName: this.entityDisplayName,
+    };
+    return doc;
+  },
+  unWrap: function unWrap(doc) {
+    if (doc.entity) {
+      doc.entity.$offlineDate = doc.modifyDate;
+    }
+    return doc.entity;
+  },
+  getEntries: function getEntries(query, options) {
```

Comment:
We seem to be mixing entries and entities in the API. Will this be confusing to users?